src/AbstractArray.php

```diff
@@ -638,4 +638,23 @@ protected function sortingKeys(array &$elements, $order = SORT_ASC, $strategy =
                 ksort($elements, $strategy);
         }
     }
+
+    /**
+     * Returns the first value that matches $func condition
+     *
+     * @param Closure $func
+     *
+     * @return mixed The resulting value
+     */
+    public function find(Closure $func)
```

Comment:
I like this idea, but I prefer to rename this method to `exists`, because it's not necessary to return a real value - it already known before this method was called.
Btw, to rename this method to `exists` we'll to have the same method name as [ArrayCollection](https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/ArrayCollection.php#L200).

What do you think?